[2024-07-09T17:52:08.017+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pesquisa_industrial_anual_empresas.workspace manual__2024-07-09T17:52:06.624471+00:00 [queued]>
[2024-07-09T17:52:08.026+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pesquisa_industrial_anual_empresas.workspace manual__2024-07-09T17:52:06.624471+00:00 [queued]>
[2024-07-09T17:52:08.027+0000] {taskinstance.py:2193} INFO - Starting attempt 1 of 1
[2024-07-09T17:52:08.042+0000] {taskinstance.py:2217} INFO - Executing <Task(PapermillOperator): workspace> on 2024-07-09 17:52:06.624471+00:00
[2024-07-09T17:52:08.048+0000] {standard_task_runner.py:60} INFO - Started process 4825 to run task
[2024-07-09T17:52:08.051+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'pesquisa_industrial_anual_empresas', 'workspace', 'manual__2024-07-09T17:52:06.624471+00:00', '--job-id', '21', '--raw', '--subdir', 'DAGS_FOLDER/pesquisa_industrial_anual_empresas/dags.py', '--cfg-path', '/tmp/tmp9be6rmo6']
[2024-07-09T17:52:08.054+0000] {standard_task_runner.py:88} INFO - Job 21: Subtask workspace
[2024-07-09T17:52:08.121+0000] {task_command.py:423} INFO - Running <TaskInstance: pesquisa_industrial_anual_empresas.workspace manual__2024-07-09T17:52:06.624471+00:00 [running]> on host 049690d4da25
[2024-07-09T17:52:08.227+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='observatorio@sistemafiea.com.br' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='pesquisa_industrial_anual_empresas' AIRFLOW_CTX_TASK_ID='workspace' AIRFLOW_CTX_EXECUTION_DATE='2024-07-09T17:52:06.624471+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-07-09T17:52:06.624471+00:00'
[2024-07-09T17:52:08.228+0000] {execute.py:83} INFO - Input Notebook:  /opt/***/dags/pesquisa_industrial_anual_empresas/notebooks/workspace.ipynb
[2024-07-09T17:52:08.229+0000] {execute.py:84} INFO - Output Notebook: /opt/***/dags/resultado/pesquisa_industrial_anual_empresas_workspace_2024-07-09.ipynb
[2024-07-09T17:52:09.923+0000] {clientwrap.py:44} INFO - Executing notebook with kernel: python3
[2024-07-09T17:52:16.744+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/papermill/operators/papermill.py", line 118, in execute
    pm.execute_notebook(
  File "/home/airflow/.local/lib/python3.11/site-packages/papermill/execute.py", line 131, in execute_notebook
    raise_for_execution_errors(nb, output_path)
  File "/home/airflow/.local/lib/python3.11/site-packages/papermill/execute.py", line 251, in raise_for_execution_errors
    raise error
papermill.exceptions.PapermillExecutionError: 
---------------------------------------------------------------------------
Exception encountered at "In [4]":
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[4], line 7
      4 response = requests.get(url)
      5 data = response.json()
----> 7 schema = StructType([
      8     StructField("id", StringType(), True),
      9     StructField("variavel", StringType(), True),
     10     StructField("unidade", StringType(), True),
     11     StructField("resultados", 
     12         StructType([
     13             StructField("classificacoes", 
     14                 StructType([
     15                     StructField("id", StringType(), True),
     16                     StructField("nome", StringType(), True),
     17                     StructField("categoria", MapType(StringType(), StringType()), True)
     18                 ]), True),
     19             StructField("series", 
     20                 StructType([
     21                     StructField("localidade", 
     22                         StructType([
     23                             StructField("id", StringType(), True),
     24                             StructField("nivel", 
     25                                 StructType([
     26                                     StructField("id", StringType(), True),
     27                                     StructField("nome", StringType(), True)
     28                                 ]), True),
     29                             StructField("nome", StringType(), True)
     30                         ]), True),
     31                     StructField("serie", MapType(StringType(), StringType()), True)
     32                 ]), True)
     33         ]), True)
     34 ])
     36 # Converta a lista de dicionários para um DataFrame usando o schema explícito
     37 rdd = spark.sparkContext.parallelize([data])

NameError: name 'StructType' is not defined

[2024-07-09T17:52:16.757+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=pesquisa_industrial_anual_empresas, task_id=workspace, execution_date=20240709T175206, start_date=20240709T175208, end_date=20240709T175216
[2024-07-09T17:52:16.772+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 21 for task workspace (
---------------------------------------------------------------------------
Exception encountered at "In [4]":
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[4], line 7
      4 response = requests.get(url)
      5 data = response.json()
----> 7 schema = StructType([
      8     StructField("id", StringType(), True),
      9     StructField("variavel", StringType(), True),
     10     StructField("unidade", StringType(), True),
     11     StructField("resultados", 
     12         StructType([
     13             StructField("classificacoes", 
     14                 StructType([
     15                     StructField("id", StringType(), True),
     16                     StructField("nome", StringType(), True),
     17                     StructField("categoria", MapType(StringType(), StringType()), True)
     18                 ]), True),
     19             StructField("series", 
     20                 StructType([
     21                     StructField("localidade", 
     22                         StructType([
     23                             StructField("id", StringType(), True),
     24                             StructField("nivel", 
     25                                 StructType([
     26                                     StructField("id", StringType(), True),
     27                                     StructField("nome", StringType(), True)
     28                                 ]), True),
     29                             StructField("nome", StringType(), True)
     30                         ]), True),
     31                     StructField("serie", MapType(StringType(), StringType()), True)
     32                 ]), True)
     33         ]), True)
     34 ])
     36 # Converta a lista de dicionários para um DataFrame usando o schema explícito
     37 rdd = spark.sparkContext.parallelize([data])

NameError: name 'StructType' is not defined
; 4825)
[2024-07-09T17:52:16.800+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-07-09T17:52:16.822+0000] {taskinstance.py:3312} INFO - 0 downstream tasks scheduled from follow-on schedule check
